contracts/TwabRewards.sol

```diff
@@ -0,0 +1,379 @@
+        internal
+        pure
+        returns (Epoch memory)
+    {
+        uint256 _epochDuration = _promotion.epochDuration;
+
+        return
+            Epoch({
+                id: uint32(_epochId),
+                startTimestamp: uint32(_promotion.startTimestamp + (_epochDuration * _epochId)),
+                duration: uint32(_epochDuration)
+            });
+    }
+
+    /**
+        @notice Get current epoch settings.
+        @return Epoch settings
+     */
+    function _getCurrentEpoch() internal view returns (Epoch memory) {
+        Promotion memory _currentPromotion = _getCurrentPromotion();
+        uint256 _numberOfEpochs = _currentPromotion.numberOfEpochs;
+
+        uint256 _promotionEndTimestamp = _currentPromotion.startTimestamp +
+            (_currentPromotion.epochDuration * _numberOfEpochs);
+        uint256 _currentEpochId = (_numberOfEpochs / block.timestamp) * _promotionEndTimestamp;
+
+        return _getEpoch(_currentEpochId, _currentPromotion);
+    }
+
+    /**
+        @notice Get reward amount for a specific user.
+        @dev Rewards can only be claimed once the epoch is over.
+        @param _user User to get reward amount for
+        @param _promotionId Promotion id from which the epoch is
+        @param _epochId Epoch id to get reward amount for
+        @return Reward amount
+     */
+    function _calculateRewardAmount(
+        address _user,
+        uint256 _promotionId,
+        uint256 _epochId
+    ) internal view returns (uint256) {
+        Promotion memory _promotion = _getPromotion(_promotionId);
+        Epoch memory _epoch = _getEpoch(_epochId, _promotion);
+
+        uint256 _epochDuration = _epoch.duration;
+        uint256 _epochStartTimestamp = _epoch.startTimestamp;
+        uint256 _epochEndTimestamp = _epochStartTimestamp + _epochDuration;
+
+        require(
+            block.timestamp >= _epochStartTimestamp && block.timestamp <= _epochEndTimestamp,
+            "TwabRewards/epoch-not-over"
+        );
+
+        ITicket _ticket = ITicket(_promotion.ticket);
+
+        uint256 _averageBalance = _ticket.getAverageBalanceBetween(
+            _user,
+            uint64(_epochStartTimestamp),
+            uint64(_epochEndTimestamp)
+        );
+
+        uint64[] memory _epochStartTimestamps = new uint64[](1);
+        _epochStartTimestamps[0] = uint64(_epochStartTimestamp);
+
+        uint64[] memory _epochEndTimestamps = new uint64[](1);
+        _epochEndTimestamps[0] = uint64(_epochEndTimestamp);
+
+        uint256[] memory _averageTotalSupplies = _ticket.getAverageTotalSuppliesBetween(
+            _epochStartTimestamps,
+            _epochEndTimestamps
+        );
+
+        uint256 _averageTotalSupply = _averageTotalSupplies[0];
+
+        /// User's share of tickets expressed in percentage
+        uint256 shareOfTickets = (_averageBalance * 100) / _averageTotalSupply;
+
+        return (_promotion.tokensPerEpoch * shareOfTickets) / 100;
```

Comment:
Two rules:

1. Do as few operations as possible
2. Always multiply before dividing.

You're trying to compute: `tokensPerEpoch * (avgBal / avgTotSupply)`

This is equivalent to `(tokensPerEpoch * avgBal) / avgTotSupply`

No need for the 100.

Reply:
Done: https://github.com/pooltogether/v4-periphery/pull/8/commits/420cda573d53755e437f173df5fb4c21e28bb3d6